### PR TITLE
Set Redis db

### PR DESCRIPTION
**Pull request by ericgaspar**
- Set Redis DB number

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

File: scripts/_common.sh [hunk 1]
```diff
@@ -20,3 +20,45 @@ NODEJS_VERSION=16
...
+	do
+	 	if ! echo "$result" | grep -q "db$i"
+	 	then
+			db=$i
+	 		break 1
+ 		fi
+ 		db=-1
+	done
+
+	test "$db" -eq -1 && ynh_die --message="No available Redis databases..."
```
Comment (fflorent) [inline scripts/_common.sh:51]: Nitpicking: more compact but maybe less easy to read (regexp can be black magic):

```bash
local max_db=$(grep -Po "^databases\s+\K\d+" /etc/redis/redis.conf)
local cur_upper_db_number=$(redis-cli INFO keyspace | grep -Po "^db\K\d+" | sort -n | tail -1)
local db=$((1+${cur_upper_db_number:=-1}))
test "$db" -lt "$max_db" || ynh_die --message="No available Redis databases..."
echo "$db"
```

The `\K` is explained on this page (chapter "Limitation of LookBehind"): https://fabianlee.org/2021/01/09/bash-grep-with-lookbehind-and-lookahead-to-isolate-desired-text/

What do you think? (don't hesitate to tell me you prefer your current proposal, I can totally understand)

**Review by fflorent — COMMENTED** [review]
Mostly good for me. The most important remark is about `$max-1` and otherwise LGTM.

File: scripts/_common.sh (hunk 1 above)
Comment (ericgaspar) [reply scripts/_common.sh:51]: Thanks for the review.
This code is part of `ynh_redis` experimental helper. https://github.com/YunoHost-Apps/Experimental_helpers/blob/master/ynh_redis/ynh_redis
For now this helper is quiet minimale. If you feel like improving Redis helper, there is a backup and restaure mecanism to add as well as many other things...

**Comment by fflorent** [comment]
Upgrade tested, all seems fine.
LGTM!

**Review by yalh76 — APPROVED** [review]

File: scripts/_common.sh [hunk 2]
```diff
@@ -20,3 +20,45 @@ NODEJS_VERSION=16
...
+ynh_redis_get_free_db() {
+	local result max db
+	result="$(redis-cli INFO keyspace)"
+
+	# get the num
+	max=$(cat /etc/redis/redis.conf | grep ^databases | grep -Eow "[0-9]+")
+
+	db=0
+	# default Debian setting is 15 databases
+	for i in $(seq 0 "$max")
```
[the remark this answers is not in the document]
Comment (yalh76) [reply scripts/_common.sh:41]: Would be better to comment at https://github.com/YunoHost-Apps/Experimental_helpers/blob/master/ynh_redis/ynh_redis

**Pull request merged**